flytekit/core/python_function_task.py

```diff
@@ -349,15 +349,20 @@ def dynamic_execute(self, task_function: Callable, **kwargs) -> Any:
         raise ValueError(f"Invalid execution provided, execution state: {ctx.execution_state}")
 
     def _write_decks(self, native_inputs, native_outputs_as_map, ctx, new_user_params):
-        # These errors are raised if the source code can not be retrieved
-        with suppress(OSError, TypeError):
-            source_code = inspect.getsource(self._task_function)
+        if self._disable_deck is False:
+            # These errors are raised if the source code can not be retrieved
+            with suppress(OSError, TypeError):
+                source_code = inspect.getsource(self._task_function)
 
-            from flytekit.deck import Deck
-            from flytekit.deck.renderer import SourceCodeRenderer
+                from flytekit.deck import Deck
+                from flytekit.deck.renderer import PythonDependencyRenderer, SourceCodeRenderer
 
-            source_code_deck = Deck("Source Code")
-            renderer = SourceCodeRenderer()
-            source_code_deck.append(renderer.to_html(source_code))
+                source_code_deck = Deck("Source Code")
+                renderer = SourceCodeRenderer()
+                source_code_deck.append(renderer.to_html(source_code))
+
+                python_dependencies_deck = Deck("Python Dependencies")
+                renderer = PythonDependencyRenderer()
+                python_dependencies_deck.append(renderer.to_html())
```

Comment:
Since the `supress` is only for `getsource`:

```python
if self._disable_deck is False:
    from flytekit.deck import Deck

    # These errors are raised if the source code can not be retrieved
    with suppress(OSError, TypeError):
        source_code = inspect.getsource(self._task_function)
        from flytekit.deck.renderer import SourceCodeRenderer

        source_code_deck = Deck("Source Code")
        renderer = SourceCodeRenderer()
        source_code_deck.append(renderer.to_html(source_code))

    python_dependencies_deck = Deck("Python Dependencies")
    renderer = PythonDependencyRenderer()
    python_dependencies_deck.append(renderer.to_html())

```

Reply:
that's a nice touch.

Reply:
good advice, thanks